### DISCUSSION TITLE
Behaviour with groups and stabilityDays

### DISCUSSION THREAD

**Discussion by geraintwhite** [renovatebot/renovate#9956] [Request Help] [answered]
I have recently added groups to my renovate config in an attempt to reduce the noise in my repository. I also use `"stabilityDays": 3` and a weekly schedule and I wondered what the behaviour will be with this combination of config.

Will the stability days apply so that every update within a group must be at least 3 days old, or so that the entire group remains unchanged for 3 days?

```json
{
  "extends": [
    "config:base",
    "group:linters",
    "schedule:weekly",
    ":automergePatch",
    ":rebaseStalePrs",
    ":noUnscheduledUpdates",
    ":prNotPending",
    ":dependencyDashboard"
  ],
  "stabilityDays": 3,
  "enabledManagers": ["npm", "nvm"],
  "node": {
    "supportPolicy": ["lts_latest"],
    "major": { "enabled": true }
  },
  "ignoreDeps": ["flow-bin", "electron"],
  "ignorePaths": ["**/node_modules/**", "**/Pods/**"],
  "packageRules": [
    {
      "groupName": ["dependencies"],
      "depTypeList": ["dependencies"],
      "updateTypes": ["patch"],
      "automerge": true
    },
    {
      "groupName": ["devDependencies"],
      "depTypeList": ["devDependencies"],
      "updateTypes": ["minor", "patch"],
      "automerge": true
    }
  ]
}
```

**Comment by viceice** [comment, marked as answer]
`stabilityDays` will apply to all updates. if you like to supress unstable releases from groups, you should set `internalChecksFilter=strict`

https://docs.renovatebot.com/configuration-options/#internalchecksfilter

**Reply by geraintwhite** [reply]
So does that mean with my current configuration a PR will be created for a group only once all deps within it are stable? And with `internalChecksFilter=strict`, unstable deps will be excluded from the group?

**Reply by viceice** [reply]
yes `internalChecksFilter=strict` will suppress any update which is not stable from further branch / pr processing

**Reply by viceice** [reply]
if you have default value of `prCreation=immediate` renovate will not wait for stabilityDays, instead it will create the pr and add a `stabilityDaysPending` check, so your branch can't be automerged.

**Reply by rarkins** [reply]
With default settings (internalChecksFilter=none), all packages will be included in the group, whether they have reached stabilityDays or not. Your `:prNotPending' will mean the PR won't get created until some point when all packages are simultaneously stable. If after the PR has been created then more releases in the group happen, then the PR will be marked as pending again but remain open.

With internalChecksFilter=strict, the goal is to filter out any which don't reach stabilityDays before they reach the group, but I haven't actually test it with a group yet to verify for sure how it behaves. You can either test it out yourself or wait a bit and I'll try to test.

**Reply by rarkins** [reply]
With default settings:

```
DEBUG: Update has not passed stability days (repository=renovate-tests/internalchecks1, branch=renovate/dependencies)
       "depName": "aws-sdk",
       "daysElapsed": 0,
       "stabilityDays": 10
DEBUG: Skipping branch creation due to stability days not met (repository=renovate-tests/internalchecks1, branch=renovate/dependencies)
```

i.e. the branch is pending and not created.

Now I'm going to test the other settings.

**Reply by geraintwhite** [reply]
Thanks for this. I'll give `internalChecksFilter=strict` a try. To be clear, does it go in each `packageRule`?

**Reply by rarkins** [reply]
I tested it out and right now it may give unexpected results when grouped, but that's something I aim to have fixed today. I would recommend once you add it, just put it at the top level of your config instead of inside every package rule.
